core/src/main/java/feast/core/job/JobUpdateTask.java

```diff
@@ -101,11 +101,19 @@ public Job call() {
     }
   }
 
-  boolean featureSetsChangedFor(Job job) {
-    Set<FeatureSet> existingFeatureSetsPopulatedByJob = Sets.newHashSet(job.getFeatureSets());
-    Set<FeatureSet> newFeatureSetsPopulatedByJob = Sets.newHashSet(featureSets);
+  boolean requiresUpdate(Job job) {
+    // If set of feature sets has changed
+    if (!Sets.newHashSet(featureSets).equals(Sets.newHashSet(job.getFeatureSets()))) {
+      return true;
+    }
 
-    return !newFeatureSetsPopulatedByJob.equals(existingFeatureSetsPopulatedByJob);
+    // If any existing feature set populated by the job has its status as pending
```

Comment:
Curious: was this addition a semantic fix independent of versions? More practical side of the question: should it be backported to v0.4?

Reply:
Related: #708, probably also/instead the backport candidate.

Reply:
Or rather, #708 obviates my question so there is no backport need.